monai/apps/auto3dseg/algorithms/templates/segresnet/scripts/infer.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import sys
+from typing import Optional, Sequence, Union
+
+import torch
+
+import monai
+from monai import transforms
+from monai.bundle import ConfigParser
+from monai.bundle.scripts import _pop_args, _update_args
+from monai.data import ThreadDataLoader, decollate_batch, list_data_collate
+from monai.inferers import sliding_window_inference
+
+
+class InferClass:
```

Comment:
Could you please help define some abstract classes in "bundle_scripts.py", similar to:
https://github.com/Project-MONAI/MONAI/blob/b4fa8860a0b40e87786b5fc7a9fdcf0f5dbcccbf/monai/apps/auto3dseg/bundle_gen.py
Then every algorithm template should inherit the base classes to implement `Infer`, `Train`, `Evaluate`,
It should clearly define the "must-have" APIs.

Thanks in advance.